x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/input/search/ExecutableSearchInput.java

```diff
@@ -86,7 +87,12 @@ public ExecutableSearchInput(SearchInput input, Client client, WatcherSearchTemp
 
         final Payload payload;
         if (input.getExtractKeys() != null) {
-            BytesReference bytes = XContentHelper.toXContent(response, XContentType.JSON, false);
+            Map<String, String> paramsMap = new HashMap<>();
```

Comment:
instead of allocating a new objects, can you juse have a static `MapParams` variable, that uses `Collections.singletonMap()`?